### ISSUE TITLE
[feat] uv support

### ISSUE THREAD

**Issue by henryiii** [pypa/cibuildwheel#1837]
### Description

I'm looking into supporting uv in cibuildwheel. From initial tests, this can roughly half the time of simple jobs.

For example, Windows:
![Screenshot 2024-05-27 at 2 06 26 AM](https://github.com/pypa/cibuildwheel/assets/4616906/582284c1-6a08-429b-a9a5-947d4af66085)
![Screenshot 2024-05-27 at 2 06 38 AM](https://github.com/pypa/cibuildwheel/assets/4616906/ca1da697-1aac-4860-b054-b330a2ee6c2c)

The total Windows time went from about 9 minutes to about 6 minutes for `deflate`.

Linux:

![Screenshot 2024-05-27 at 2 03 41 AM](https://github.com/pypa/cibuildwheel/assets/4616906/a6e1f335-c42b-4f38-bb88-40a6539433d1)
![Screenshot 2024-05-27 at 2 03 17 AM](https://github.com/pypa/cibuildwheel/assets/4616906/918f0113-cc56-4314-abe6-a46e97d1df81)

(I don't have the WIP working well enough to finish on Linux for a total time, or running on macOS)

**Comment by henryiii** [comment]
PS: This is on hold waiting for a version of uv with the 32-bit linux fix.

**Comment by mayeut** [comment]
@henryiii, do you have more info to share on the 32-bit linux issue ? I did not find anything in the uv repo.

**Comment by henryiii** [comment]
Waiting on this one to be in a release: https://github.com/astral-sh/uv/pull/3899

Without it it can't find a 32-bit Python.

**Comment by mayeut** [comment]
I opended https://github.com/astral-sh/uv/pull/3963 for `ppc64le`

**Comment by henryiii** [comment]
Released. Should be possible to continue on this then when manylinux updates.

**Comment by mayeut** [comment]
The new version is in for x86_64/i686 images. Other images are in the queue on Travis CI

**Comment by henryiii** [comment]
![image](https://github.com/pypa/cibuildwheel/assets/4616906/e56658cf-79a1-42ab-8e45-455d7d993b23)

![image](https://github.com/pypa/cibuildwheel/assets/4616906/91fed9d0-aeec-4a0b-979e-dd6af4abe599)

**Comment by henryiii** [comment]
In 2.19!
